Invalid homeDir for export command

The wasmd doesn't pass the homeDir properly when using export command.
It passes always `wasm` as the homeDir.

**Step to reproduce**
```
wasmd export

                                                                          1 ↵
panic: Error calling the VM: Cache error: Error opening Wasm file for reading: No such file or directory (os error 2)

goroutine 1 [running]:
github.com/CosmWasm/wasmd/x/wasm/internal/keeper.ExportGenesis.func1(0x1, 0xc000fc5a80, 0x20, 0x20, 0xc000f6b950, 0x2b, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/CosmWasm/wasmd/x/wasm/internal/keeper/genesis.go:67 +0x2b4
github.com/CosmWasm/wasmd/x/wasm/internal/keeper.Keeper.IterateCodeInfos(0x5bc7de0, 0xc001068210, 0x5c00260, 0xc000c80470, 0x5bc7de0, 0xc001068140, 0x5bf9c40, 0xc000c80470, 0x5bf9c40, 0xc000c80470, ...)
	github.com/CosmWasm/wasmd/x/wasm/internal/keeper/keeper.go:574 +0x287
github.com/CosmWasm/wasmd/x/wasm/internal/keeper.ExportGenesis(0x5be7be0, 0xc0000c8030, 0x5bfd860, 0xc0010809c0, 0x0, 0x0, 0x0, 0x0, 0x116, 0x0, ...)
```


![debugging](https://user-images.githubusercontent.com/2441640/101757073-10160880-3b1a-11eb-8b88-5d76b4a52ad9.jpg)

--- a/cmd/wasmd/root.go
+++ b/cmd/wasmd/root.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"context"
+	"errors"
 	"io"
 	"os"
 	"path/filepath"
@@ -108,7 +109,7 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig app.EncodingConfig) {
 		debug.Cmd(),
 	)
 
-	server.AddCommands(rootCmd, app.DefaultNodeHome, newApp, createWasnAppAndExport, addModuleInitFlags)
+	server.AddCommands(rootCmd, app.DefaultNodeHome, newApp, createWasmAppAndExport, addModuleInitFlags)
 
 	// add keybase, auxiliary RPC, query, and tx child commands
 	rootCmd.AddCommand(
@@ -237,19 +238,23 @@ func newApp(logger log.Logger, db dbm.DB, traceStore io.Writer, appOpts serverty
 	)
 }
 
-func createWasnAppAndExport(
+func createWasmAppAndExport(
 	logger log.Logger, db dbm.DB, traceStore io.Writer, height int64, forZeroHeight bool, jailAllowedAddrs []string,
 	appOpts servertypes.AppOptions) (servertypes.ExportedApp, error) {
 
 	var wasmApp *app.WasmApp
+	homePath, ok := appOpts.Get(flags.FlagHome).(string)
+	if !ok || homePath == "" {
+		return servertypes.ExportedApp{}, errors.New("application home not set")
+	}
 	if height != -1 {
-		wasmApp = app.NewWasmApp(logger, db, traceStore, false, map[int64]bool{}, "", uint(1), app.GetEnabledProposals(), appOpts)
+		wasmApp = app.NewWasmApp(logger, db, traceStore, false, map[int64]bool{}, homePath, uint(1), app.GetEnabledProposals(), appOpts)
 
 		if err := wasmApp.LoadHeight(height); err != nil {
 			return servertypes.ExportedApp{}, err
 		}
 	} else {
-		wasmApp = app.NewWasmApp(logger, db, traceStore, true, map[int64]bool{}, "", uint(1), app.GetEnabledProposals(), appOpts)
+		wasmApp = app.NewWasmApp(logger, db, traceStore, true, map[int64]bool{}, homePath, uint(1), app.GetEnabledProposals(), appOpts)
 	}
 
 	return wasmApp.ExportAppStateAndValidators(forZeroHeight, jailAllowedAddrs)

